### DISCUSSION TITLE
Is there any way to run termux under proot? (or chroot)

### DISCUSSION THREAD

**Discussion by JesusChapman** [termux/termux-app#2906] [Q&A]
It's just a question, I don't feel very comfortable with docker, since when opening a terminal I must always wait a while for the dns to be updated, it always does, I would like to run it in proot or chroot, it seems more comfortable, is it possible?

**Comment by Spz2022** [comment]
Maybe yes

**Reply by JesusChapman** [reply]
How?

**Reply by TomJo2000** [reply]
You should be able to do a local SSH connection into Termux from proot.

Be aware, you're still going to be incurring the performance overhead of proot.

**Reply by sylirre** [reply]
> Maybe yes

Maybe no if look more deeply into the problem. Without Android OS system root, you cannot use Termux packages. Without Android init system you cannot have DNS - described in known issues of https://github.com/termux/termux-docker.

> You should be able to do a local SSH connection into Termux from proot.

That does not answer the original question: *I don't feel very comfortable with docker, since when opening a terminal I must always wait a while for the dns to be updated, it always does, I would like to run it in proot or chroot, it seems more comfortable, is it possible?*

However the idea of SSH and Android device running Termux will be a suitable solution if OP is not comfortable using image from https://github.com/termux/termux-docker.

**Reply by Spz2022** [reply]
https://github.com/z80oolong/proot-termux-build

**Comment by Maiwandamany** [comment]
I need termux package and python install

در تاریخ شنبه ۱۹ نوامبر ۲۰۲۲،‏ ۸:۴۵ ق.ظ Spz2022 ***@***.***>
نوشت:
